realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -373,25 +373,18 @@ public void moveLastOver(long rowIndex) {
         nativeMoveLastOver(nativePtr, rowIndex);
     }
 
-    public long addEmptyRow() {
+    /**
+     * Add an empty row to the table.
+     *
+     * @param checkPrimaryKey set to true to check if this table has a primary key defined. Throw an exception if yes.
+     * @return row index.
+     * @throws RealmException if row cannot be added.
+     */
+    public long addEmptyRow(boolean checkPrimaryKey) {
         checkImmutable();
-        if (hasPrimaryKey()) {
-            long primaryKeyColumnIndex = getPrimaryKey();
-            RealmFieldType type = getColumnType(primaryKeyColumnIndex);
-            switch (type) {
-                case STRING:
-                    if (findFirstString(primaryKeyColumnIndex, STRING_DEFAULT_VALUE) != NO_MATCH) {
-                        throwDuplicatePrimaryKeyException(STRING_DEFAULT_VALUE);
-                    }
-                    break;
-                case INTEGER:
-                    if (findFirstLong(primaryKeyColumnIndex, INTEGER_DEFAULT_VALUE) != NO_MATCH) {
-                        throwDuplicatePrimaryKeyException(INTEGER_DEFAULT_VALUE);
-                    }
-                    break;
-                default:
-                    throw new RealmException("Cannot check for duplicate rows for unsupported primary key type: " + type);
-            }
+        if (!checkPrimaryKey && hasPrimaryKey()) {
```

Comment:
In order to throw, both subexpressions must be true, right? So `checkPrimaryKey` must be false. This is not consistent with doc above. Or what am I missing :-)